### DISCUSSION TITLE
Add `Packed*Array(<elements>)` literals to GDScript

### DISCUSSION THREAD

**Discussion by dalexeev** [godotengine/godot-proposals#5653] [Scripting]
GDScript does not support `PackedInt32Array(1, 2, 3)` syntax, only `PackedInt32Array([1, 2, 3])`[^1]. At the same time, `.tscn`, `.tres`, `var_to_str`, etc. do not use `[]`.

```gdscript
print(var_to_str(PackedInt32Array([1, 2, 3]))) # PackedInt32Array(1, 2, 3)
```

[^1]: Does this mean that the `Array` is created first, and then it is converted to `Packed*Array`?
Possibly related: godotengine/godot#67873.

**Comment by Calinou** [comment]
This might be implementable, but only if Variant constructors support using `vararg`. The performance difference of using this on medium/large-sized arrays should also be checked.

**Reply by dalexeev** [reply]
Even if `Variant` doesn't support it, we could recognize it at the GDScript parser/tokenizer (`Packed*Array` literals).
